tools/tpm2_encryptdecrypt.c

```diff
@@ -99,11 +99,11 @@ static bool encrypt_decrypt(TSS2_SYS_CONTEXT *sapi_context) {
         },
     };
 
-    TPM_RC rval = Tss2_Sys_EncryptDecrypt(sapi_context, ctx.key_handle,
-            &sessions_data, ctx.is_decrypt, TPM_ALG_NULL, &iv_in, &ctx.data, &out_data,
+    TPM_RC rval = Tss2_Sys_EncryptDecrypt2(sapi_context, ctx.key_handle,
+            &sessions_data, &ctx.data, ctx.is_decrypt, TPM_ALG_NULL, &iv_in, &out_data,
             &iv_out, &sessions_data_out);
```

Comment:
I would either look at the supported commands via getcapability or try version 2 than default to 1. I can work on fixing this up and getting the newer tpm simulator supported.

Reply:
As a first pass I'll go with the later. I've got the fix for #469 in here now. Will do this bit tomorrow and get a PR together. Thanks for the feedback.